### PR TITLE
[IMP] pos_*,sale: improve success page

### PR DESCRIPTION
**Pull request by davidmonnom**
*: pos_self_order,pos_online_payment_self_order,pos_restaurant

Commit 1:
Before, when ordering on the kiosk/mobile, no confirmation page was
displayed, only a notification.

Now, a confirmation page displays the tracking number, the amount paid
and, if present, the customer's table..

---
Commit 2:
The problem is that a mandatory field from the sale module has been
added to the ProductAttributeCustomValue template, but in point of sale
we also use this model without necessarily filling this field.

Previously, the sale module was the only module to use the
ProductAttributeCustomValue model (product.attribute.custom.value).
In the override, a field set to mandatory has been added
"sale_order_line_id", a relationship between the sale.order.line model
and product.attribute.custom.value.

We now remove the mandatory parameter from this field so that we can
create new data without getting an error indicating that a mandatory
field is missing. This has no impact on the operation of the Sale module
as this field will be populated anyway, since the only way to create a
record of this type is via a sale_order_line or a pos_order_line.

---
Previously, when a table was selected on a floor plan other than the
first, clicking on back brought you to the first floor plan, not the
one of the previously selected table.

Now, when you click on the back button, you return to the floor plan
corresponding to the previous table.

Also, the order number is now displayed on the floor screen button.
image

### CODE REVIEW

**Review by Feyensv — APPROVED** [review]

File: addons/pos_self_order/static/src/app/pages/confirmation_page/confirmation_page.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    get showPayMessage() {
+        const config = this.selfOrder.config;
+        const paymentMethods = this.selfOrder.pos_payment_methods.filter((p) =>
+            config.self_ordering_mode === "mobile" ? p.is_online_payment : !p.is_online_payment
+        );
+
+        return paymentMethods.length === 0 && config.self_ordering_pay_after === "each";
+    }
```
Comment (adgu-odoo) [inline addons/pos_self_order/static/src/app/pages/confirmation_page/confirmation_page.js:86]: Is this used ?

File: addons/pos_self_order/static/src/app/pages/landing_page/landing_page.js [hunk 2]
```diff
@@ -57,6 +57,29 @@ export class LandingPage extends Component {
         return "";
     }
 
+    get draftOrder() {
+        return this.selfOrder.orders.filter((o) => o.access_token && o.state === "draft");
+    }
```
Comment (adgu-odoo) [inline addons/pos_self_order/static/src/app/pages/landing_page/landing_page.js:62]: You are only using this getter to get the length of the draft orders. Maybe return the length in the getter and not the orders

File: addons/pos_self_order/static/src/app/pages/landing_page/landing_page.xml [hunk 3]
```diff
@@ -22,17 +22,22 @@
...
+                        </t>
                     </a>
                 </t>
             </div>
             <div t-if="selfOrder.custom_links.length" class="d-flex gap-3 flex-grow-1">
                 <t t-foreach="selfOrder.custom_links" t-as="link" t-key="link.id">
                     <a type="button"
                         t-on-click="(event) => this.clickCustomLink(link)"
-                        t-attf-class="btn btn-lg btn-{{link.style}}">
+                        t-attf-class="btn btn-lg btn-{{link.style}} {{ hideBtn(link) ? 'd-none' : ''}}">
```
Comment (adgu-odoo) [inline addons/pos_self_order/static/src/app/pages/landing_page/landing_page.xml:40]: To hide a button, use a t-if rather that a d-none added to the class

File: addons/pos_self_order/static/src/app/pages/landing_page/landing_page.js [hunk 4]
```diff
@@ -57,6 +57,29 @@ export class LandingPage extends Component {
...
+    hideBtn(link) {
+        const arrayLink = link.url.split("/");
+        const routeName = arrayLink[arrayLink.length - 1];
+
+        if (routeName !== "products") {
+            return;
+        }
+
+        return (
+            routeName === "products" &&
```
Comment (adgu-odoo) [inline addons/pos_self_order/static/src/app/pages/landing_page/landing_page.js:73]: You do not need this part of the condition

**Review by adgu-odoo — CHANGES_REQUESTED** [review]
Nice PR

File: addons/pos_online_payment_self_order/static/src/pages/cart_page/cart_page.js [hunk 5]
```diff
@@ -19,26 +19,35 @@ patch(CartPage.prototype, {
...
+            return;
+        }
+
         if (!this.selfOrder.table && service === "table" && !takeAway) {
             this.state.selectTable = true;
             return;
         }
 
-        if (mode === "meal" && isOnlinePayment) {
+        if (mode === "meal" && isOnlinePayment && order.isSavedOnServer) {
```
[the remark this answers is not in the document]
Comment (davidmonnom) [reply addons/pos_online_payment_self_order/static/src/pages/cart_page/cart_page.js:31]: In fact, it's the second check that's useless, so I've deleted it.

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**